pyspark_ai/tool.py

```diff
@@ -101,13 +103,70 @@ async def _arun(self, *args: Any, **kwargs: Any) -> Any:
         raise NotImplementedError("ListTablesSqlDbTool does not support async")
 
 
+class LRUVectorStore:
+    """Implements an LRU policy to enforce a max storage space for vector file storage."""
+
+    def __init__(self, vector_file_dir: str, max_size: float = 1e6) -> None:
+        # by default, max_size = 1e6 GB
+        self.files: OrderedDict[str, float] = OrderedDict()
+        self.vector_file_dir = vector_file_dir
+        # represent in bytes to prevent floating point errors
+        self.max_bytes = max_size * 1e9
+        self.current_size = 0
+
+        # initialize the file cache, if vector_file_dir exists
+        # existing files will get evicted in reverse-alphabetical order
+        # TODO: write LRU to disk, to evict existing files in LRU order
+        if os.path.exists(self.vector_file_dir):
+            for file in os.listdir(self.vector_file_dir):
+                file_full_path = os.path.join(self.vector_file_dir, file)
+                file_size = os.path.getsize(file_full_path)
```

Comment:
we should either use `os.path.getsize` in all the code, or all use `LRUVectorStore.get_file_size_gb` in all the code.